platform/ext/target/nordic_nrf/common/nrf5340/target_cfg.c

```diff
@@ -712,11 +717,19 @@ enum tfm_plat_err_t spu_periph_init_cfg(void)
      * - TWISx
      * - UARTEx
      */
+
+    /* UART0 is a secure peripheral, so we need to leave Serial-Box 0 as Secure.
+     * The UART Driver will configure it as non-secure when it uninitializes.
+     */
+#if !(defined(SECURE_UART1) && defined(CONFIG_TFM_SHARE_UART))
```

Comment:
Since we always require that SECURE_UART1 is not defined it is better to have a top if statement with this and a nested statement for the other configurations.

Reply:
You're missing the parenthesis, the not is for the whole expression. So your suggestion is not possible without duplicating the call to "config_non_secure".